crates/bevy_transform/src/components/transform.rs

```diff
@@ -314,7 +314,8 @@ impl Transform {
     /// If this [`Transform`] has a parent, the `axis` is relative to the rotation of the parent.
     #[inline]
     pub fn rotate_axis(&mut self, axis: Dir3, angle: f32) {
-        self.rotate(Quat::from_axis_angle(axis.into(), angle));
+        // Normalize rotation due to potential for exponential denormalization (#16480)
+        self.rotation = (Quat::from_axis_angle(axis.into(), angle) * self.rotation).normalize();
```

Comment:
Why not just add the normalization on a second line instead of re-implementing rotate?

```rust
self.rotate(Quat::from_axis_angle(axis.into(), angle));
// Normalize rotation due to potential for exponential denormalization (#16480)
self.rotation = self.rotation.normalize();
```

Reply:
Done! Had it like that at first, then arbitrarily decided against it :P